Assets/Scripts/Framework/GeoLocation/RouteHighlighter.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Framework.GeoLocation
+{
+    [RequireComponent(typeof(LineRenderer))]
+    public sealed class RouteHighlighter : MonoBehaviour
+    {
+        private const float INVOKE_DELAY = 0.1f;
+
+        [SerializeField] private UnityEvent withinRange = new UnityEvent();
+        
+        [SerializeField] private List<Transform> routePoints;
+        [SerializeField] private LineRenderer lineRenderer;
+        [SerializeField] private float closeRange = 5f;
+        
+        [SerializeField] private Transform _nextPoint;
+
+        private void Start()
+        {
+            Invoke(nameof(InitLine), INVOKE_DELAY);
+            _nextPoint = routePoints[1];
+            
+        }
+
+        private void Update()
+        {
+            if (Vector3.Distance(transform.position, _nextPoint.position) < closeRange)
+            {
+                withinRange?.Invoke();
+            }
+        }
+
+        public void RemoveLine()
+        {
+            if (routePoints.Count <= 2)
```

Comment:
```suggestion
            if (routePoints.Count < 2)
```
Als er nog 2 punten zijn dan gaat de line weg. Maar de 2 punten zijn dan de speler en het laatste punt waar je naartoe gaat lopen, dat weet je niet meer waar je naartoe moet als het lang lopen is.